File: Modul 6/PRAK604_2510817120002_AmaliaSoraya.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void identifikasi(int panjang, char *pesan1, char *pesan2) {
    char simbolis[100];
    int count_bintang = 0, count_tagar = 0;

    for(int i = 0; i < panjang-1; i++) {
        if(pesan1[i] == pesan2[i] && pesan2[i] == ' ') {
            simbolis[i] = ' ';
        } else if(pesan1[i] == pesan2[i]) {
            simbolis[i] = '*';
            count_bintang++;
        } else if(pesan1[i] != pesan2[i]) {
            simbolis[i] = '#';
            count_tagar++;
        }
    }
    printf("\n");
    for(int k = 0; k < panjang-1; k++) {
        printf("%c", simbolis[k]);
    }

    printf("\n* = %d", count_bintang);
    printf("\n# = %d", count_tagar);
    
    if(count_bintang >= count_tagar) {
        printf("\nPesan asli");
    } else {
        printf("\nPesan palsu");
    }
}

int main() {
    char pesan1[100], pesan2[100];

    fgets(pesan1, sizeof(pesan1), stdin);
    fgets(pesan2, sizeof(pesan2), stdin);

    int panjang = strlen(pesan1);

    if (strlen(pesan1) != strlen(pesan2)) {
        printf("Panjang kalimat berbeda, pesan palsu");
    } 
    
    else {
        identifikasi(panjang, pesan1, pesan2);
    }
    return 0;
}

File: Modul 6/PRAK604_2510817120002_AmaliaSoraya.py
pesan1 = input()
pesan2 = input()

def identifikasi(pesan1, pesan2) :
    simbolis = []
    bintang = 0
    tagar = 0

    for i in range(len(pesan1)) :
        if pesan1[i] == pesan2[i] and pesan2[i] == " " :
            simbolis.append(" ")
        elif pesan1[i] == pesan2[i] :
            simbolis.append("*")
            bintang += 1
        elif pesan1[i] != pesan2[i] :
            simbolis.append("#")
            tagar += 1
            
    print("\n")
    for simbol in simbolis :
        print(simbol, end="")

    print(f"\n* = {bintang}")
    print(f"# = {tagar}")
    print(f"Pesan asli" if bintang >= tagar else "Pesan palsu")

if len(pesan1) != len(pesan2) :
    print("Panjang kalimat berbeda, pesan palsu")

else :
    identifikasi(pesan1, pesan2)
